eclipse-theia/theia
# [vscode] semantic highlighting support

#### What it does


- fix #7608: porting VS Code support of  semantic highlighting
- [x] CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22677 (waiting for check-in approval)

#### How to test


- start theia
- open application-shell.ts
- observe how imports are colored before ts smartness is ready and after
  - you can see whether ts smartness initialization progress in the status bar
- open preferences side by side with the editor, find semantic highligting toggle, use it and see how highlighting is changing in the editor

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
